src/contexts/RestaurantListContext.jsx

```diff
@@ -0,0 +1,28 @@
+import { createContext, useState, useEffect } from "react";
+import { getRestaurantList } from "../api/restaurantAPI";
+
+export const RestaurantListContext = createContext();
+
+export function RestaurantListProvider({ children }) {
+  const [restaurantList, setRestaurantList] = useState([]);
+
+  useEffect(() => {
+    getRestaurantList().then((data) => {
+      setRestaurantList(data);
+    });
+  }, []);
```

Comment:
```suggestion
  useEffect(() => {
   updateRestaurantList();
  }, []);
```
아래에 함수로 updateRestaurantList를 정의했으니 여기서도 불러와서 사용하면 좋을 거 같은데 예은님의 생각은 어떤가요..?!

Reply:
지금 보니 그렇게 하니까 중복되는 코드를 줄일 수 있네요! 고치겠습니다